pkg/registry/client.go

```diff
@@ -0,0 +1,177 @@
+package registry
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strings"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras-go/v2"
+	orasregistry "oras.land/oras-go/v2/registry"
+	"oras.land/oras-go/v2/registry/remote"
+	"oras.land/oras-go/v2/registry/remote/auth"
+)
+
+// OCIRegistryClient storage client for an OCI registry.
+type OCIRegistryClient struct {
+	host        string
+	project     string
+	certFile    string
+	insecure    bool
+	dryRun      bool
+	initialized bool
+	OI          OrasInterface
+	registry    *remote.Registry
+}
+
+var _ StorageClient = (*OCIRegistryClient)(nil)
+
+// NewOCIRegistry create an OCI registry client.
+func NewOCIRegistry(host string, certFile string, insecure bool) *OCIRegistryClient {
+	return &OCIRegistryClient{
+		host:     host,
+		certFile: certFile,
+		insecure: insecure,
+		OI:       &OrasImplementation{},
+	}
+}
+
+// Init registry configuration.
+func (or *OCIRegistryClient) Init() error {
+	if or.initialized {
```

Comment:
I guess there's not much to worry about, but this is a race condition... Consider a `sync.Mutex` or a `sync.Once` if it's really important that this code only be run once per instantiation. I don't see anything here that isn't safe to run more than once?